warc_metadata_sidecar.py

```diff
@@ -0,0 +1,277 @@
+                continue
+            payload = io.BytesIO(record.content_stream().read())
+            # The payload is how we find the important info. Skip record if empty.
+            if not payload.read(1):
+                continue
+            url = record.rec_headers.get_header('WARC-Target-URI')
+            record_date = record.rec_headers.get_header('WARC-Date')
+            warcinfo_id = record.rec_headers.get_header('WARC-Warcinfo-ID')
+            warcrecord_id = record.rec_headers.get_header('WARC-Record-ID')
+            # Define specific warc_headers to include in sidecar.
+            warc_dict = {'WARC-Date': record_date, 'WARC-Concurrent-ID': warcrecord_id}
+            if warcinfo_id:
+                warc_dict['WARC-Warcinfo-ID'] = warcinfo_id
+
+            print(url)
+            payload.seek(0)
+            mime_and_puid = find_mime_and_puid(fido, payload)
+            result_dict = {}
+            lang_cld = None
+            if mime_and_puid[0] is not {}:
+                if mime_and_puid[0].get('fido') and mime_and_puid[0].get('python-magic'):
+                    record_mime = "r'(" + mime_and_puid[0].get('fido') + "|" + mime_and_puid[0].get('python-magic') + ")'"
+                else:
+                    if mime_and_puid[0].get('fido'):
+                        record_mime = mime_and_puid[0].get('fido')
+                    else:
+                        record_mime = mime_and_puid[0].get('python-magic')
+                # Check that the format matches either mime type or just the one.
+                if TEXT_FORMAT_MIMES.search(record_mime):
+                    payload.seek(0)
+                    bytes_payload = payload.read()
+                    result_dict = find_character_set(bytes_payload)
+                    lang_cld = find_language(bytes_payload)
+                    text_mime += 1
+                else:
+                    non_text += 1
+            string_payload = create_string_payload(mime_and_puid, result_dict, lang_cld)
+            if not string_payload:
+                continue
+            record_count += 1
+            meta_record = writer.create_warc_record(url,
+                                                    'metadata',
+                                                    payload=io.BytesIO(string_payload.encode()),
+                                                    warc_headers_dict=warc_dict
+                                                    )
+            writer.write_record(meta_record)
+        # Delete sidecar file if we do not collect any records.
+        if not record_count:
+            os.remove(warc_file_path)
+            logging.info('Deleted sidecar, no records to collect.')
+        else:
+            logging.info('Finished creating sidecar in %s',
+                         str(timedelta(seconds=(time.time() - start))))
+            logging.info('Found %s response/resource record(s)', record_count)
```

Comment:
It seems like `record_count` isn't necessarily the number of response/resource records found and checked, but the count of ones that some sort of mimetype, puid, language, or encoding was found for? Or does that always come out to the same thing?

Reply:
`record_count` is the number of records we write to the sidecar, which would only be `response` or `resource` records and yes they are the ones that have some sort of mime type, puid, language, and/or encoding.

Reply:
My confusion is with the logged text saying to me that this is the number of all response/records _found_ as in the count of every response/resource record appearing in the original WARC vs. these are the number of response/records that we determined some type/language info on. If it is possible that some response/records are found, but no sidecar record is written for them, then I think the text would be more clear if it said something like "Determined sidecar information for %s response/resource record(s)"